packages/worker/src/workflow/reusable-vm.ts

```diff
@@ -104,39 +119,62 @@ export class ReusableVMWorkflowCreator implements WorkflowCreator {
    */
   async createWorkflow(options: WorkflowCreateOptions): Promise<Workflow> {
     const context = this.context;
-    const bag: Record<string, unknown> = {};
-    const { isolateExecutionTimeoutMs, contextKeysToPreserve } = this;
+    const pristine = this.pristine;
+    const bag: Map<string, PropertyDescriptor> = new Map();
+    const { isolateExecutionTimeoutMs } = this;
     const workflowModule: WorkflowModule = new Proxy(
       {},
       {
         get(_: any, fn: string) {
           return (...args: any[]) => {
-            Object.assign(context, bag);
+            for (const [pname, pdesc] of bag.entries()) {
```

Comment:
nit: would have liked to prevent constructing the entries array for every time we call into the VM.